test/master-playlist-controller.test.js

```diff
@@ -324,6 +324,59 @@ QUnit.test('fast quality change resyncs audio segment loader', function(assert)
   assert.equal(resets, 0, 'does not reset the audio segment loader when media changes');
 });
 
+QUnit.test('resets everything for a fast quality change', function(assert) {
+  let resyncs = 0;
+  let resets = 0;
+  let removeFuncArgs = {};
+
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+  // master
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
+
+  let segmentLoader = this.masterPlaylistController.mainSegmentLoader_;
+
+  segmentLoader.resyncLoader = () => resyncs++;
+
+  segmentLoader.remove = function(start, end) {
+    removeFuncArgs = {
+      start,
+      end
+    };
+  };
+
+  segmentLoader.duration_ = () => 60;
+
+  segmentLoader.on('reseteverything', function() {
+    resets++;
+  });
+
+  // media is unchanged
+  this.masterPlaylistController.fastQualityChange_();
+
+  assert.equal(resyncs, 0, 'does not resync segment loader if media is unchanged');
+
+  assert.equal(resets, 0, 'reseteverything event not triggered if media is unchanged');
+
+  assert.deepEqual(removeFuncArgs, {}, 'remove() not called if media is unchanged');
+
+  // media is changed
+  this.masterPlaylistController.selectPlaylist = () => {
+    return this.masterPlaylistController.master().playlists[0];
+  };
+
+  this.masterPlaylistController.fastQualityChange_();
+
+  assert.equal(segmentLoader.ended_, false, 'segment loader ended property is false');
```

Comment:
Was there a specific reason for the ended check?

Reply:
I included it since `ended_` gets set to `false` as a side effect of calling `resetEverything()`. If you think it's superfluous I can remove it.

Reply:
Probably don't need it here. While it's a decent test, since these tests are meant for MasterPlaylistController we can probably leave checking that state change within the SegmentLoader tests.